paimon-core/src/test/java/org/apache/paimon/disk/InMemoryBufferTest.java

```diff
@@ -143,4 +147,70 @@ public void testEmpty() throws Exception {
         RowBufferIterator iterator = buffer.newIterator();
         assertThat(iterator.advanceNext()).isFalse();
     }
+
+    @Test
+    public void testMemoryPoolWorksWellWithInMemoryBuffer() throws Exception {
```

Comment:
There are lots of code style issues, you can check your ide support.

Reply:
done